compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -2184,20 +2184,34 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 let suffix = chars.collect::<String>();
                 suffix.is_empty() || suffix == "f32" || suffix == "f64"
             };
+            let is_likely_suffix = |fist_chars: &[char], field: &str| {
+                field.len() >= 1
+                    && field.to_lowercase().starts_with(fist_chars)
+                    && field[1..].chars().all(|c| c.is_ascii_digit())
+            };
             if let ty::Infer(ty::IntVar(_)) = expr_t.kind()
                 && let ExprKind::Lit(Spanned {
                     node: ast::LitKind::Int(_, ast::LitIntType::Unsuffixed),
                     ..
                 }) = base.kind
                 && !base.span.from_expansion()
-                && is_valid_suffix(field_name)
             {
-                err.span_suggestion_verbose(
-                    field.span.shrink_to_lo(),
-                    "If the number is meant to be a floating point number, consider adding a `0` after the period",
-                    '0',
-                    Applicability::MaybeIncorrect,
-                );
+                let msg = "If the number is meant to be a floating point number, consider adding a `0` after the period";
```

Comment:
```suggestion
                let msg = "if a floating point literal was intended, consider adding a `0` after the period";
```
pre-existing, but diagnostics shouldn't start with uppercase letters